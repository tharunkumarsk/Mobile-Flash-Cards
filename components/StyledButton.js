import React from "react";
import PropTypes from "prop-types";
import {
  Text,
  View,
  TouchableOpacity,
  StyleSheet,
  Platform
} from "react-native";
import Colors from "../constants/Colors";

export default function StyledButton({
  children,
  onPress,
  BtnStyle = "",
  disabled = true
}) {
  const BtnStyles = styles[BtnStyle];
  return (
    <View style={styles.btnContainer}>
      <TouchableOpacity
        style={[
          BtnStyles,
          Platform.OS === "ios" ? styles.iosSubmitBtn : styles.AndroidSubmitBtn
        ]}
        disabled={disabled}
        onPress={onPress}
      >
        <Text style={styles.submitBtnText}>{children}</Text>
      </TouchableOpacity>
    </View>
  );
}
const styles = StyleSheet.create({
  iosSubmitBtn: {
    padding: 10,
    textAlign:"center",
    borderRadius: 7,
    height: 50,
   
  },
  AndroidSubmitBtn: {
    padding: 10,
    paddingLeft: 30,
    paddingRight: 30,
    height: 45,
    borderRadius: 2,
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "stretch"
  },
  submitBtnText: {
    color: Colors.white,
    fontSize: 22,
    textAlign: "center"
  },
  btnContainer: {
    paddingBottom: 15
  },
  btnDisabled: {
    backgroundColor: Colors.gray,
    borderColor: Colors.gray,
    color: Colors.gray
  },
  btnPrimary: {
    backgroundColor: Colors.green
  },
  btnSecondary: {
    backgroundColor: Colors.red
  },
  btnTeritary: {
    backgroundColor: Colors.purple
  }
});

StyledButton.propTypes = {
  children: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  BtnStyle: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired
};
