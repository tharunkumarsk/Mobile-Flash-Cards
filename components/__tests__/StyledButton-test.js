import React from 'react';
import renderer from 'react-test-renderer';

import StyledButton  from '../StyledButton';

it(`renders correctly the primary button`, () => {
  const tree = renderer.create(<StyledButton
    onPress={ handleSubmit = () => {}}
    BtnStyle = "btnPrimary"
    >
    Snapshot test!</StyledButton>).toJSON();

  expect(tree).toMatchSnapshot();
});
it(`renders correctly the secondary button`, () => {
  const tree = renderer.create(<StyledButton
    onPress={ handleSubmit = () => {}}
    BtnStyle = "btnSecondary"
    >
    Snapshot test!</StyledButton>).toJSON();

  expect(tree).toMatchSnapshot();
});
it(`renders correctly the disabled button`, () => {
  const tree = renderer.create(<StyledButton
    onPress={ handleSubmit = () => {}}
    BtnStyle = "btnDisabled"
    >
    Snapshot test!</StyledButton>).toJSON();

  expect(tree).toMatchSnapshot();
});
