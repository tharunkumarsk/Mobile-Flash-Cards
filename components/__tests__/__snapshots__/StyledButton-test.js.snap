// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the disabled button 1`] = `
<View
  style={
    Object {
      "paddingBottom": 15,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#757575",
        "borderColor": "#757575",
        "borderRadius": 7,
        "color": "#757575",
        "height": 45,
        "marginLeft": 40,
        "marginRight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
      Snapshot test!
    </Text>
  </View>
</View>
`;

exports[`renders correctly the primary button 1`] = `
<View
  style={
    Object {
      "paddingBottom": 15,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00965e",
        "borderRadius": 7,
        "height": 45,
        "marginLeft": 40,
        "marginRight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
      Snapshot test!
    </Text>
  </View>
</View>
`;

exports[`renders correctly the secondary button 1`] = `
<View
  style={
    Object {
      "paddingBottom": 15,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#b71845",
        "borderRadius": 7,
        "height": 45,
        "marginLeft": 40,
        "marginRight": 40,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 22,
          "textAlign": "center",
        }
      }
    >
      Snapshot test!
    </Text>
  </View>
</View>
`;
